src/components/AuthWrappers/PrivatePageWrapper.tsx

```diff
@@ -0,0 +1,40 @@
+import { useRouter } from 'next/router'
+import { useMemo, type PropsWithChildren } from 'react'
+import { CALLBACK_URL_KEY } from '~/constants/params'
+import { useAuth } from '~/features/auth'
+import { SIGN_IN } from '~/lib/routes'
+import { FullscreenSpinner } from '../FullscreenSpinner'
+
+interface PrivatePageWrapperProps {
+  /**
+   * Route to redirect to when user is not authenticated. Defaults to
+   * `SIGN_IN` route if not provided.
+   */
+  redirectTo?: string
+}
+
+const Redirect = ({ redirectTo }: PrivatePageWrapperProps) => {
+  const router = useRouter()
+  const redirectUrl = useMemo(() => {
+    if (typeof window === 'undefined') return encodeURIComponent('/')
```

Comment:
noob question from me: why do we need to check if `typeof window === 'undefined'` here? Thanks :)

Reply:
ah, cause this component may be invoked on the server side if using server-side rendering, but in that case there should be a server-side redirect already and this component does not need to be used. The check is just a safeguard against crashing when `window` is referenced on the server.